File: db/src/main/java/com/delivery/db/userordermenu/UserOrderMenuRepository.java
package com.delivery.db.userordermenu;

import com.delivery.db.userordermenu.enums.UserOrderMenuStatus;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserOrderMenuRepository extends JpaRepository<UserOrderMenuEntity, Long> {

    List<UserOrderMenuEntity> findAllByUserOrderIdAndStatus(Long userOrderId,
        UserOrderMenuStatus status);

}

File: db/src/main/kotlin/com/delivery/db/userordermenu/UserOrderMenuRepository.kt
package com.delivery.db.userordermenu

import com.delivery.db.userordermenu.enums.UserOrderMenuStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserOrderMenuRepository : JpaRepository<UserOrderMenuEntity, Long> {
    fun findAllByUserOrderIdAndStatus(
        userOrderId: Long?,
        status: UserOrderMenuStatus?
    ): List<UserOrderMenuEntity>
}
